app/models/product.rb

```diff
@@ -10,7 +11,26 @@ class Product
   field :version, type: String
   field :description, type: String
 
-  validates :name, presence: true
+  validates :name, presence: true, uniqueness: { scope: :vendor, message: "Product name was already taken. Please choose another." }
+  validates :ehr_type, presence: true
+  validate :at_least_one_test_type?
   validates :vendor, presence: true
-  # methods must be added
+
+  field :ehr_type, type: String
+  field :c1_test, type: Boolean
```

Comment:
the purpose of the booleans is to give the choice to the user who makes the product for which types they will be testing for. ex.) if only c1_test is true then all measure tests should include the ability to only test c1 for that measure